Week-5/Assignment-1/index.js

```diff
@@ -0,0 +1,69 @@
+function* generator() {
+  while (true) {
+    let obj = {
+      userId: Math.floor(Math.random() * 1000),
+    };
+    yield obj;
+  }
+}
+
+async function getUser(userId) {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve({
+        userId: userId,
+        username: (Math.random() + 1).toString(36).substring(7),
+      });
+    }, 1000);
+  });
+}
+
+async function getServices(service) {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve(service);
+    }, 2000);
+  });
+}
+
+async function getServiceCost(services) {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      resolve(services.length * 100);
+    }, 3000);
+  });
+}
+
+async function getDetails(uId) {
+  let userDetails = {};
+
+  await getUser(uId)
+    .then((res) => {
```

Comment:
in case of async await why are you using then()? await getUser(uId) should return the resolved value